File: personal-hub-server/ph-knowledge/src/main/java/com/personalhub/knowledge/entity/Category.java
package com.personalhub.knowledge.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 统一分类实体
 */
@Data
@TableName("category")
public class Category {

    /** 主键 */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 所属用户ID */
    private Long userId;

    /** 分类名称 */
    private String name;

    /** 分类类型: note, bookmark, file */
    private String type;

    /** 排序 */
    private Integer sortOrder;

    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /** 更新时间 */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: personal-hub-web/src/types/category.ts
export interface CategoryVO {
  id: number
  name: string
  type: 'note' | 'bookmark' | 'file'
  sortOrder: number
  count: number
  createdAt: string
}

export interface CategoryCreateDTO {
  name: string
  type: 'note' | 'bookmark' | 'file'
  sortOrder?: number
}

export interface CategoryUpdateDTO {
  name: string
  sortOrder?: number
}
